### PR TITLE
fix(core): resolve Valid workflow issues

### LINKED ISSUE #42: When user create the incomplete workflow and save it displays error msg.
**Describe the bug**
When user create the incomplete workflow and save it displays error msg.

For Further information
https://sourcefuse.atlassian.net/browse/OPX-9005

**Expected behavior**
It should be valid Workflow when saves 
**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by Deepika516**
resolve issues so that it can create valid workflow

GH-42

## Description
Do this Condition & When this happens condition is not optional so without this the workflow cannot be saved.workflow should be valid when saved .

Fixes #42 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Intermediate change (work in progress)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist:

- [ ] Performed a self-review of my own code
- [ ] npm test passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the style guide
- [ ] API Documentation in code was updated
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by yeshamavani — APPROVED** [review]

**Review by AryanshSourcefuse — APPROVED** [review]

File: projects/workflows-creator/src/lib/builder/builder.component.ts [hunk 1]
```diff
@@ -462,8 +469,52 @@ export class BuilderComponent<E> implements OnInit, OnChanges {
...
+        switch (node.node.getIdentifier()) {
+          case EventTypes.OnChangeEvent:
+          case EventTypes.OnValueEvent:
+          case ActionTypes.ChangeColumnValueAction:
+            const columnExists = !!node.node.state.get('column');
+            const valueExists = !!node.node.state.get('value');
+            const valueTypeIsAnyValue =
+              node.node.state.get('valueType') === ValueTypes.AnyValue;
+            isValid = columnExists && (valueExists || valueTypeIsAnyValue);
+            break;
```
Comment (akshatdubeysf) [inline projects/workflows-creator/src/lib/builder/builder.component.ts:492]: Shouldn't this be part of the respective node classes?

**Review by akshatdubeysf — CHANGES_REQUESTED** [review]

File: projects/workflows-creator/src/lib/builder/builder.component.ts (hunk 1 above)
Comment (Deepika516) [reply projects/workflows-creator/src/lib/builder/builder.component.ts:492]: yes it should be , can u plz tell me the approach which i should follow sir, but sir we need to give this in emergency so

**Review by akshatdubeysf — APPROVED** [review]

File: projects/workflows-creator/src/lib/builder/builder.component.ts (hunk 1 above)
Comment (Deepika516) [reply projects/workflows-creator/src/lib/builder/builder.component.ts:467]: sure sir ,I will create the seprate issue of this .

**Comment by yeshamavani** [comment]
@Deepika516  create a new issue and link it here

**Comment by Deepika516** [comment]
#54

**Pull request merged**